### PR TITLE
NH-5775 trace context in sql 2 - postgresql and refactorings

### PR DESCRIPTION
**Pull request by prettycoder**
This PR is mainly about adding trace context to postgresql queries see sequel.rb
and a few other refactorings following from this:

- add trace context for sequel postgresql queries (incl tests)
- add QueryTag kv (incl tests)
- remove traceparent when reporting Query kv
- add kvs arg as reference to add_traceparent_to_sql
- refactor assigning kvs to use kvs arg as reference
- fix traces index in tests as kvs are now added to the exit event
- fix rails instrumentation and sql tests for the kvs arg change
- use latest postgres in github actions
- replace kill with pkill
- remove unused travis file

### CODE REVIEW

File: lib/appoptics_apm/util.rb [hunk 1]
```diff
@@ -156,8 +156,17 @@ def to_query(h)
...
+        sql.gsub(/\\\'/,'').gsub(@@regexp, '?')
+      end
+
+      ##
+      # remove_traceparent
+      #
+      # Remove trace context injection
+      #
+      def remove_traceparent(sql)
+        sql.gsub(AppOpticsAPM::SDK::CurrentTraceInfo::TraceInfo::SQL_REGEX, '')
```
Comment (ronilan) [inline lib/appoptics_apm/util.rb:169]: I'd call it `remove_sql_tag`

File: lib/appoptics_apm/frameworks/rails/inst/connection_adapters/utils5x.rb [hunk 2]
```diff
@@ -50,50 +48,45 @@ def trace_wrap(*args)
...
-              # Sanitize SQL and don't report binds
-              kvs[:Query] = AppOpticsAPM::Util.sanitize_sql(sql)
-            else
-              # Report raw SQL and any binds if they exist
-              kvs[:Query] = sql.to_s
-              if binds && !binds.empty?
-                kvs[:QueryArgs] = binds.map(&:value)
-              end
+        def assign_kvs(sql, kvs, name = nil, binds = [])
+          sql = AppOpticsAPM::Util.remove_traceparent(sql.to_s)
```
Comment (ronilan) [inline lib/appoptics_apm/frameworks/rails/inst/connection_adapters/utils5x.rb:52]: `# make sure query kv does not include injected sql tag.`

File: lib/appoptics_apm/inst/sequel.rb [hunk 3]
```diff
@@ -69,17 +69,52 @@ def extract_trace_details(sql, opts)
...
         end
+      end
 
-        send(method, sql, opts, &block)
-      rescue => e
-        AppOpticsAPM::API.log_exception(:sequel, e)
-        raise e
-      ensure
-        AppOpticsAPM::API.log_exit(:sequel)
+      def add_traceparent(sql, kvs)
```
Comment (ronilan) [inline lib/appoptics_apm/inst/sequel.rb:80]: I'd call it `inject_sql_tag`

File: lib/appoptics_apm/inst/sequel.rb [hunk 4]
```diff
@@ -69,17 +69,52 @@ def extract_trace_details(sql, opts)
...
+          new_sql = add_traceparent_to_ps(sql, kvs)
+          return new_sql # related query may have been modified
+        end
+        sql # return original when none of the cases match
+      end
+
+      # this method uses some non-api methods partially copied from
+      # `execute_prepared_statement` in `mysql2.rb`
+      # and `prepare` in `prepared_statements.rb` in the sequel gem
+      def add_traceparent_to_ps(ps, kvs)
```
Comment (ronilan) [inline lib/appoptics_apm/inst/sequel.rb:108]: I'd call it `inject_sql_tag_into_prepared_statement`

File: lib/appoptics_apm/sdk/current_trace_info.rb [hunk 5]
```diff
@@ -76,11 +75,18 @@ def for_sql
         ##
         # add_traceparent_to_sql
         #
-        # returns the sql with "/*traceparent='#{@trace_id}'*/" prepended
+        # returns the sql with "/*traceparent='#{@tracestring}'*/" prepended
+        # and adds the QueryTag kv to kvs
         #
-        def add_traceparent_to_sql(sql)
+        def add_traceparent_to_sql(sql, kvs)
```
Comment (ronilan) [inline lib/appoptics_apm/sdk/current_trace_info.rb:81]: I'd call it `inject_sql_tag_into_sql`

File: lib/appoptics_apm/sdk/current_trace_info.rb [hunk 6]
```diff
@@ -76,11 +75,18 @@ def for_sql
...
           sql = sql.gsub(SQL_REGEX, '') # remove if it was added before
-          "#{AppOpticsAPM::SDK.current_trace_info.for_sql}#{sql}"
+
+          unless for_sql.empty?
+            kvs[:QueryTag] = for_sql
```
Comment (ronilan) [inline lib/appoptics_apm/sdk/current_trace_info.rb:85]: We ok with "side effect" modifying `kvs`?

**Review by ronilan — APPROVED** [review]
All I have to sat is about naming. You call.

File: lib/appoptics_apm/util.rb (hunk 1 above)
Comment (prettycoder) [reply lib/appoptics_apm/util.rb:169]: I'm sticking with 'add', 'remove', and 'traceparent', just names, they work fine

File: lib/appoptics_apm/frameworks/rails/inst/connection_adapters/utils5x.rb (hunk 2 above)
Comment (prettycoder) [reply lib/appoptics_apm/frameworks/rails/inst/connection_adapters/utils5x.rb:52]: I'll add later

File: lib/appoptics_apm/sdk/current_trace_info.rb (hunk 6 above)
Comment (prettycoder) [reply lib/appoptics_apm/sdk/current_trace_info.rb:85]: yes, Ruby is not functional

**Pull request merged**